text/0000-roadmap-2019.md

```diff
@@ -0,0 +1,328 @@
+- Extracting parts of rustc into **libraries**, which are easier to understand and maintain but also help in developing a Rust specification
+- Supporting lang team in **key feature development**, such as around `async fn` support, or improvements to the trait system 
+- Improving our **documentation** and **organization**, helping to make it clearer what is going on at any given time, and how people can get involved.
+
+Some of the specific projects we have in mind are as follows:
+
+- **Parallelizing rustc**, which should help with improving raw compilation times and also increasing IDE responsiveness. 
+- Introducing **MIR optimizations**, which will improve the quality of our generated code, but also may help reduce compilation time by giving LLVM less work to do (LLVM remains a large percentage of overall compilation time).
+- Introducing **working groups** for each project, and track them on the [compiler-team repository](https://github.com/rust-lang/compiler-team).
+
+One project worth calling out in more detail is the [**RLS 2.0
+effort**](https://github.com/rust-lang/compiler-team/tree/master/working-groups/rls-2.0).
+The existing RLS, while functional, was never deeply integrated into rustc
+itself. Over the past two years, we have rewritten the back end of rustc to
+use an **incremental and demand-driven infrastructure.** To get the IDE
+experience we truly want, we need to see that work through for the front end
+of the compiler as well. However, this part of rustc is one of its oldest,
+and there are significant design questions involved in how best to do that.
+
+This is where the RLS 2.0 effort comes in. **The plan is to build a prototype
+of the new front-end, thus enabling a truly first-class IDE experience.**
+Wherever possible, we aim to share code between this front-end and rustc by
+extracting libraries, such as [Chalk] (for traits), [Polonius] (for the
+borrow checker), and a new library focusing on name resolution. Eventually,
+we will merge this new front-end with the remaining bits of back-end from
+rustc.
+
+[Chalk]: https://github.com/rust-lang-nursery/chalk
+[Polonius]: http://github.com/rust-lang-nursery/polonius
+[rust-analyzer project]: https://github.com/rust-analyzer/rust-analyzer
+
+## Crates.io
+
+One of the newer teams, the Crates.io Team is going to be focused on growing
+itself, paying down technical debt in the codebase, and establishing themes
+for long-term priorities.
+
+## Dev Tools
+
+The Dev Tools team has a few core functions:
+
+- liasing with the wider project
+- being a bit of a "commons" space for cross-cutting tools concerns
+- finding out what needs are yet to be addressed by tooling and filling them
+- providing some kind of quorum helping smaller tool subteams make decisions
+
+They'll be working to fulfill these functions this year, along with looking
+for ways to improve their internal processes.
+
+Individual devtools subteams may be coming up with their own roadmaps, you
```

Comment:
How many dev-tools subteams do we want to enumerate here? For example, [i wrote out the roadmap for the Rustdoc Team over here](https://quietmisdreavus.net/code/2019/02/19/rustdoc-in-2019/#the-rustdoc-2019-roadmap), which was summarized in the [All-Hands recap post for the dev-tools team](https://internals.rust-lang.org/t/all-hands-recap-tools-team/9546).

Reply:
If you add it to the devtools repo I think we can list it here

Reply:
I'm not sure! It can't hurt to have it in the text, I think. I'm open to opinions!

Reply:
Yeah. We can inline the text or just put it in the devtools repo and link it, either works for me.

Reply:
I think putting it in the dev-tools repo and linking it here is probably best; since the IDEs roadmap is linked externally, adding a similar mention to rustdoc won't be too much of a stretch.

Reply:
Update: The Rustdoc 2019 roadmap is now live [on the Dev-Tools Team repo](https://github.com/rust-dev-tools/dev-tools-team/blob/master/roadmaps/rustdoc-2019.md). I don't know if you wanted to post an edit yourself, or if you wanted me to post a PR to add it to this.

Reply:
(The more the merrier, is my opinion)

Reply:
This still needs to be added btw

Reply:
https://github.com/steveklabnik/rfcs/pull/5